apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/internal/json/FileUploadAwareJsonWriter.kt

```diff
@@ -0,0 +1,65 @@
+package com.apollographql.apollo3.api.internal.json
+
+import com.apollographql.apollo3.api.Upload
+
+class FileUploadAwareJsonWriter(private val wrappedWriter: BufferedSinkJsonWriter): JsonWriter {
```

Comment:
wonder if `JsonWriter ` will it be more concise if we delegate to it? And just override value(Upload)

Reply:
I think that won't work because it will return the `wrappedWriter` in chained calls instead of `this`. We might remove the chained calls altogether though since we rarely use them and Kotlin makes it easy to use `apply` if needed